sql/item_geofunc.h

```diff
@@ -979,6 +979,65 @@ class Item_func_issimple: public Item_long_func_args_geometry
   { return get_item_copy<Item_func_issimple>(thd, this); }
 };
 
+
+class Item_func_latlongfromgeohash : public Item_real_func {
+ private:
+  const double lower_latitude= -90.0;
+  const double upper_latitude= 90.0;
+  const double lower_longitude= -180.0;
+  const double upper_longitude= 180.0;
```

Comment:
In MySQL, these boundary limits are defined in the [constructor function](https://github.com/mysql/mysql-server/blob/824e2b4064053f7daf17d7f3f84b7a3ed92e5fb4/sql/item_geofunc.h#L647-L656):
```
  Item_func_latlongfromgeohash(const POS &pos, Item *a, double lower_latitude,
                               double upper_latitude, double lower_longitude,
                               double upper_longitude,
                               bool start_on_even_bit_arg)
      : Item_real_func(pos, a),
        lower_latitude(lower_latitude),
        upper_latitude(upper_latitude),
        lower_longitude(lower_longitude),
        upper_longitude(upper_longitude),
        start_on_even_bit(start_on_even_bit_arg) {}
```
But they are using exactly same values in [the two subclasses](https://github.com/mysql/mysql-server/blob/824e2b4064053f7daf17d7f3f84b7a3ed92e5fb4/sql/item_geofunc.h#L669-L693).
```
  Item_func_latfromgeohash(const POS &pos, Item *a)
      : Item_func_latlongfromgeohash(pos, a, -90.0, 90.0, -180.0, 180.0, false) {}
...
  Item_func_longfromgeohash(const POS &pos, Item *a)
      : Item_func_latlongfromgeohash(pos, a, -90.0, 90.0, -180.0, 180.0, true) {
  }
```
Now seems you decide to make them hardcoded in this parent class Item_func_latlongfromgeohash and not configurable in subclasses. If that's the case, maybe better to use `static constexpr` instead?

Reply:
Exactly, I looked into their code and found out all the values are always the same and wanted to make it hardcoded, I have added the static constexpr now.